src/external_cli/cargo/metadata.rs

```diff
@@ -41,59 +41,71 @@ pub struct Metadata {
     /// List of all packages in the workspace.
     ///
     /// It also includes all feature-enabled dependencies unless `--no-deps` is used.
-    packages: Vec<Package>,
+    pub packages: Vec<Package>,
     /// List of members of the workspace.
     ///
     /// Each entry is the Package ID for the package.
-    workspace_members: Option<Vec<String>>,
+    pub workspace_members: Option<Vec<String>>,
     /// List of default members of the workspace.
     ///
     /// Each entry is the Package ID for the package.
-    workspace_default_members: Option<Vec<String>>,
+    pub workspace_default_members: Option<Vec<String>>,
     /// The absolute path to the build directory where Cargo places its output.
-    target_directory: PathBuf,
+    pub target_directory: PathBuf,
     /// The absolute path to the root of the workspace.
-    workspace_root: Option<PathBuf>,
+    pub workspace_root: Option<PathBuf>,
 }
 
 #[derive(Debug, Deserialize)]
 pub struct Package {
     /// The name of the package.
-    name: String,
+    pub name: String,
     /// The version of the package.
-    version: Version,
+    pub version: Version,
     /// The Package ID for referring to the package within the document and as the `--package`
     /// argument to many commands.
-    id: String,
+    pub id: String,
     /// List of Cargo targets.
-    targets: Vec<Target>,
+    pub targets: Vec<Target>,
     /// Absolute path to this package's manifest.
-    manifest_path: PathBuf,
+    pub manifest_path: PathBuf,
     /// Optional string that is the default binary picked by cargo run.
-    default_run: Option<String>,
+    pub default_run: Option<String>,
+}
+
+impl Package {
+    /// Check if the package has an executable binary.
+    pub fn has_bin(&self) -> bool {
+        self.targets.iter().any(|target| {
+            target
+                .kind
+                .iter()
+                .any(|target_kind| *target_kind == TargetKind::Bin)
```

Comment:
```suggestion
                .any(|target_kind| matches!(*target_kind, TargetKind::Bin))
```

This could also be the `matches!()` macro, but I like both solutions! (Yours is probably clearer, though, so let's keep that.)